lib/wireprotocol/shared.js

```diff
@@ -75,11 +79,23 @@ function isMongos(server) {
   return false;
 }
 
+function databaseNamespace(ns) {
```

Comment:
I like these functions. I am wondering if maybe we should have a single `parseNamespace` function that returns an object with `database` and `collection` fields.

Reply:
I like it.